mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/salesforce/CachedBasicConfiguration.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2020, Mulesoft, LLC. All rights reserved.
+ * Use of this source code is governed by a BSD 3-Clause License
+ * license that can be found in the LICENSE.txt file.
+ */
+package com.mulesoft.tools.migration.library.mule.steps.salesforce;
+
+import com.mulesoft.tools.migration.library.tools.SalesforceConstants;
+import com.mulesoft.tools.migration.step.AbstractApplicationModelMigrationStep;
+import com.mulesoft.tools.migration.step.ExpressionMigratorAware;
+import com.mulesoft.tools.migration.step.category.MigrationReport;
+import com.mulesoft.tools.migration.step.util.XmlDslUtils;
+import com.mulesoft.tools.migration.util.ExpressionMigrator;
+import org.jdom2.Element;
+import org.jdom2.Namespace;
+
+import java.util.Optional;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+/**
+ * Migrate BatchJob component
+ *
+ * @author Mulesoft Inc.
+ * @since 1.0.0
+ */
+public class CachedBasicConfiguration extends AbstractApplicationModelMigrationStep implements ExpressionMigratorAware {
+
+  private static final String MULE3_NAME = "cached-basic-config";
+  private static final String MULE4_CONFIG = "sfdc-config";
+  private static final String MULE4_NAME = "basic-connection";
+  private static final String MULE4_PROXY = "proxy-configuration";
+
+  private
+
+  ExpressionMigrator expressionMigrator;
+
+  public CachedBasicConfiguration() {
+    this.setAppliedTo(XmlDslUtils.getXPathSelector(SalesforceConstants.MULE3_SALESFORCE_NAMESPACE_URI, MULE3_NAME, false));
+    this.setNamespacesContributions(newArrayList(SalesforceConstants.MULE3_SALESFORCE_NAMESPACE));
+  }
+
+  @Override
+  public void execute(Element mule3CachedBasicConfig, MigrationReport report) throws RuntimeException {
+    getApplicationModel().addNameSpace(SalesforceConstants.MULE4_SALESFORCE_NAMESPACE,
+                                       SalesforceConstants.MULE4_SALESFORCE_NAMESPACE_URI, mule3CachedBasicConfig.getDocument());
+
+    Element mule4Config = new Element(MULE4_CONFIG, SalesforceConstants.MULE4_SALESFORCE_NAMESPACE);
+    setAttributes(mule3CachedBasicConfig, mule4Config, report);
+  }
+
+  private void setAttributes(Element mule3CachedBasicConfig, Element mule4Config, MigrationReport report) {
+    //salesforce:sfdc-config
+    String nameValue = mule3CachedBasicConfig.getAttributeValue("name");
+    if (nameValue != null) {
+      mule4Config.setAttribute("name", nameValue);
+    } else {
+      report.report("salesforce.cachedBasicConfiguration", mule4Config, mule4Config);
```

Comment:
We avoid adding an entry to the report because a field wasn't present.
We add entries to the report on whether we cannot perform a migration because there is no way to migrate it or they could use something else to achieve an action.